### PR TITLE
fix(api/DialogPlugin): Allow string type as component param to custom component

### PR DESCRIPTION
**Pull request by soulsam480**
**What kind of change does this PR introduce?** 

- [X] Bugfix
- [ ] Feature
- [ ] Documentation
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** 

- [ ] Yes
- [X] No

**The PR fulfills these requirements:**

- [X] It's submitted to the `dev` branch (or `v[X]` branch)
- [X] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] It's been tested on a Cordova (iOS, Android) app
- [ ] It's been tested on a Electron app
- [ ] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar/tree/dev/docs)  or explained in the PR's description.

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to [start a new feature discussion](https://github.com/quasarframework/quasar/discussions/new?category=ideas-proposals) first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by soulsam480** [comment]
Related to https://github.com/quasarframework/quasar/issues/8885. While that fix allowed  `string` as component type in runtime, the TS type definitions were not allowing the same. This PR intends to fix that.

File: ui/src/plugins/Dialog.json [hunk 1]
```diff
@@ -307,7 +307,7 @@
             },
 
             "component": {
-              "type": "Component",
+              "type": ["Component","String"],
```
Comment (yusufkandemir) [inline ui/src/plugins/Dialog.json]: To match with the existing code style.
```suggestion
              "type": [ "Component", "String" ],
```

**Review by yusufkandemir — APPROVED** [review]
👍

**Comment by yusufkandemir** [comment]
Thanks for noticing that and contributing 👏 Here are some notes/recommendations that can help on your future contributions:


The build was not passing:
```
[Error] build.api.js: src/plugins/Dialog.json ->  "methods"/"create"/"params"/"opts"/"definition"/"component" missing required API prop "examples" for its type (MultipleTypes)

{
  type: [ 'Component', 'String' ],
  desc: "Use custom dialog component; use along with 'componentProps' prop where possible"
}
```
Please see https://github.com/quasarframework/quasar/blob/dev/CONTRIBUTING.md for information about how to get a development setup, so you can build, test stuff yourself, etc.

So, I made the required changes and got a successful build. Here is the diff output the build gives:

 📜 Changes for dist/types/index.d.ts:
```diff
@@ -13730,9 +13730,9 @@
   transitionHide?: string;
   /**
    * Use custom dialog component; use along with 'componentProps' prop where possible
    */
-  component?: Component;
+  component?: Component | string;
   /**
    * User defined props which will be forwarded to underlying custom component if 'component' prop is used
    */
   componentProps?: any;
```

So, everything is looking correct now. Which means it can be merged 🎉

**Comment by sambitevidev** [comment]
Hi, thanks for the comments. Will look into the guide and follow the same in future contributions.

**Pull request merged**